File: Sources/FitUI/Components/FitRoleTag.swift
import SwiftUI

// MARK: - FitRoleTag
//
// Compact corner badge labelling the user role context of an event:
// "Athlete" (dumbbell glyph) or "Coach" (whistle glyph). Used by
// FitCalEvent in cross-role state — placed at the bottom-right corner
// to signal "this event lives on your OTHER role profile."
//
// 18pt high, 10pt font, 99pt radius. Subtle bg (white 0.08 dark / black
// 0.05 light) — meant to be unobtrusive vs. the tile content.

public struct FitRoleTag: View {
    let role: FitRole

    @Environment(\.fitTheme) private var theme

    public init(role: FitRole) {
        self.role = role
    }

    public var body: some View {
        HStack(spacing: 4) {
            Image(systemName: iconName)
                .font(.system(size: 10, weight: .medium))
                .foregroundColor(theme.textTertiary)
            Text(label)
                .font(.custom(FitFont.family, size: 10).weight(.medium))
                .foregroundColor(theme.textTertiary)
        }
        .padding(.horizontal, 8)
        .frame(height: 18)
        .background(tagBackground)
        .clipShape(Capsule())
    }

    private var iconName: String {
        switch role {
        case .athlete: return "figure.run"
        case .coach:   return "person.fill"
        }
    }

    private var label: String {
        switch role {
        case .athlete: return "Athlete"
        case .coach:   return "Coach"
        }
    }

    /// Subtle wash — slightly different alpha per theme so it reads on both
    /// dark surface-high and light gray-100 cross-role tile backgrounds.
    private var tagBackground: Color {
        // White wash on dark / black wash on light.
        // Hardcoded alphas mirror the prototype's rgba(255,255,255,0.08) /
        // rgba(0,0,0,0.05).
        Color.primary.opacity(0.05)
    }
}

File: android/src/main/kotlin/com/fit321/fitui/components/FitRoleTag.kt
package com.fit321.fitui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.DirectionsRun
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.fit321.fitui.theme.LocalFitTheme
import com.fit321.fitui.tokens.FitFont

// ============================================================================
// FitRole — user role concept shared across FitUI (matches Swift `FitRole`).
// ============================================================================

enum class FitRole { Athlete, Coach }

// ============================================================================
// FitRoleTag — small corner badge labelling the user role context.
//
// Used by FitCalEvent in cross-role state — placed at the bottom-right of the
// tile to signal "this event lives on your OTHER role profile."
// 18dp tall, 10sp font, capsule. Subtle wash, deliberately unobtrusive.
// ============================================================================

@Composable
fun FitRoleTag(
    role: FitRole,
    modifier: Modifier = Modifier
) {
    val theme = LocalFitTheme.current
    val tagBg = Color.Black.copy(alpha = 0.05f)   // subtle on both themes

    Row(
        modifier = modifier
            .clip(CircleShape)
            .background(tagBg)
            .padding(horizontal = 8.dp)
            .height(18.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        Icon(
            imageVector = when (role) {
                FitRole.Athlete -> Icons.Filled.DirectionsRun
                FitRole.Coach   -> Icons.Filled.Person
            },
            contentDescription = null,
            tint = theme.textTertiary,
            modifier = Modifier.size(10.dp)
        )
        Text(
            text = when (role) {
                FitRole.Athlete -> "Athlete"
                FitRole.Coach   -> "Coach"
            },
            style = FitFont.caption.copy(fontSize = 10.sp, fontWeight = FontWeight.Medium),
            color = theme.textTertiary
        )
    }
}
